### PR TITLE
feat(messaging, apple): allow system to display button for in-app notification settings

### PR DESCRIPTION
**Pull request by StanleyCocos**
## Description

I found that there is no option to set a button to display the app settings in the notification settings when registering for push notifications. I hope to add this feature.

fixes https://github.com/firebase/flutterfire/issues/13688

Specifically, the UNAuthorizationOptions.UNAuthorizationOptionProvidesAppNotificationSettings provided by iOS is missing.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by Lyokone** [comment]
Hey 👋 Thanks a lot for the PR, and sorry for the late review, it looks really good.

**Review by Lyokone — APPROVED** [review]

File: packages/firebase_messaging/firebase_messaging/example/lib/permissions.dart [hunk 1]
```diff
@@ -57,9 +57,9 @@ class _Permissions extends State<Permissions> {
     return Container(
       margin: const EdgeInsets.only(bottom: 8),
       child: Row(
-        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        // mainAxisAlignment: MainAxisAlignment.spaceBetween,
```
Comment (Lyokone) [inline packages/firebase_messaging/firebase_messaging/example/lib/permissions.dart:60]: Can you remove this line? I think it's better with Expanded 👍

Comment (StanleyCocos) [reply packages/firebase_messaging/firebase_messaging/example/lib/permissions.dart:60]: Thank you  I will delete this code.

**Comment by StanleyCocos** [comment]
Thank you Already submitted

**Comment by StanleyCocos** [comment]
@Lyokone   
Thank you
What else do I need to do to merge this request?

**Comment by Lyokone** [comment]
Hello @StanleyCocos, a lot of CI are currently red, you would need to apply proper formatting to your PR (`melos run format-ci`) then check the different actions to get it merged

**Comment by StanleyCocos** [comment]
Thank you. 
I will revise and follow up in time.

**Comment by StanleyCocos** [comment]
@Lyokone 
Hello, are you still continuing with the review?

**Comment by StanleyCocos** [comment]
@russellwheatley Could you please guide me? It seems that my request isn't getting approved due to CI issues. I'm unsure if any adjustments are still needed on my part. 
Thank you

**Comment by StanleyCocos** [comment]
It seems like everyone is quite busy. However, I still hope to get some help. What I know is that there are CI issues, but I’m unsure how to resolve them. Can I get some guidance? I’m following up almost every day.

**Comment by russellwheatley** [comment]
@StanleyCocos - looks good to me. I've approved CI run, let's see what the result is 👍

**Comment by StanleyCocos** [comment]
@russellwheatley 
Thank you for your response. I’ve adjusted the test, but there still seems to be an e2e issue, and I’m not sure what to do. Could you provide some guidance?

**Comment by russellwheatley** [comment]
@StanleyCocos - I've just merged with main branch which ought to have a fix for CI. Nothing on your end to do now 🙏

**Comment by StanleyCocos** [comment]
Thank you. 
It seems there’s still an issue. To be honest, I might not have a deep understanding of CI. If you could provide some direction, I would really appreciate it.

**Review by russellwheatley — APPROVED** [review]

**Review by Lyokone — APPROVED** [review]

**Pull request merged**